aiida/engine/processes/calcjobs/tasks.py

```diff
@@ -370,12 +407,13 @@ async def execute(self):  # pylint: disable=invalid-overridden-method
                 raise RuntimeError('Unknown waiting command')
 
         except TransportTaskException as exception:
-            raise plumpy.PauseInterruption(f'Pausing after failed transport task: {exception}')
-        except plumpy.KillInterruption:
+            raise plumpy.process_states.PauseInterruption(f'Pausing after failed transport task: {exception}')
+        except plumpy.process_states.KillInterruption:
             await self._launch_task(task_kill_job, node, transport_queue)
-            self._killing.set_result(True)
+            if self._killing is not None:  # TODO possible source of bug
+                self._killing.set_result(True)
```

Comment:
This fix needs checking

Reply:
that's an interesting catch. 

Does this mean that when this is being encountered, somewhere (in the logs?) there should be an error message like
```python
In [4]: class A():
   ...:     def __init__(self):
   ...:         self._killing = None
   ...:
   ...:     def b(self):
   ...:         return self._killing.set_result(True)
   ...:
   ...:

In [5]: a=A()

In [6]: a.b()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-6-4f53a1908077> in <module>
----> 1 a.b()

<ipython-input-4-3b0911a3e651> in b(self)
      4
      5     def b(self):
----> 6         return self._killing.set_result(True)
      7
      8

AttributeError: 'NoneType' object has no attribute 'set_result'
```
`set_result` [is not mentioned in the aiida issues](https://github.com/aiidateam/aiida-core/issues?q=is%3Aissue+is%3Aopen+set_result) but I guess one could still check whether any of the [kill issues](https://github.com/aiidateam/aiida-core/issues?q=is%3Aissue+is%3Aopen+kill) are resolved by this.

Reply:
P.S. Bingo

```
aiida-dev_aiida.log.4:AttributeError: 'NoneType' object has no attribute 'set_result'
aiida-dev_aiida.log.4:AttributeError: 'NoneType' object has no attribute 'set_result'
aiida-dev_aiida.log.4:AttributeError: 'NoneType' object has no attribute 'set_result'
```

Reply:
how did you get that to happen @ltalirz?

Reply:
See above. It seems to have happened in a child process being killed by a parent process (probably following a manual `verdi process kill` command).

Reply:
so possibly the cause of #3776 and #4298

Reply:
Yes, nice.  It's _probably_ a bit of a problem if the `future` is `None` so could be logged as a warning but with concurrency it's difficult to think through all the possible paths that could lead to that situation.

Reply:
added a log warning (8151a83)